packages/core/lib/commands/debug/run.js

```diff
@@ -1,21 +1,50 @@
 module.exports = async function (options) {
+  const fs = require("fs/promises");
   const { promisify } = require("util");
   const debugModule = require("debug");
   const loadConfig = require("../../loadConfig");
   const debug = debugModule("lib:commands:debug");
 
+  const Config = require("@truffle/config");
   const { Environment } = require("@truffle/environment");
   const { CLIDebugger } = require("../../debug");
+  const CompileCommon = require("@truffle/compile-common");
+  const Codec = require("@truffle/codec");
 
   if (options.url && options.network) {
     throw new Error("Url and Network options should not be specified together");
   }
 
-  let config = loadConfig(options);
+  let config;
+  let compilations;
+  try {
+    config = loadConfig(options);
+  } catch {
+    config = Config.default();
+    config.network = "development";
+    config.networks["development"] = {
+      url: "http://127.0.0.1:8545",
+      network_id: "*"
+    };
+
+    require(`${config.working_directory}/node_modules/hardhat/register`);
```

Comment:
Not needed if you then import hardhat

Reply:
I tried it without this and this line was necessary... I think I'm going to go with the approach of shelling out to `node -r hardhat/register` instead of this.